src/components/IconButton/IconButton.tsx

```diff
@@ -21,6 +21,9 @@ export interface IconButtonProps extends NativeAttributes<'button'>, Pick<BoxPro
   /** The color scheme of the button */
   variantColor?: keyof Theme['colors'];
 
+  /** The color of the icon for the 'unstyled' variant */
+  variantIconColor?: keyof Theme['colors'];
```

Comment:
Hmm I'm not sure I like that the `variantIconColor` is only applied to the "unstyled" variant. This is a bit unexpected for me as a consumer of this lib.

I think we should either do it for all styles or for none :/

Reply:
Fixed it! Now it applies to all variants. And I added a demo / test for it